test/lib/generators/cable_ready/channel_generator_test.rb

```diff
@@ -0,0 +1,125 @@
+    remove_sample_app
+  end
+
+  test "should generate channel with the same resource name and stimulus controller" do
+    run_generator ["user", "--broadcast-to", "--resource=user", "--stimulus"]
+
+    assert_file "app/channels/user_channel.rb" do |content|
+      assert_match /class\ UserChannel\ \</, content
+      assert_match /stream_for\ User\.find\(params\[\:id\]\)/, content
+    end
+
+    assert_file "app/javascript/channels/user_channel.js", /"UserChannel"/
+    assert_file "app/javascript/controllers/user_controller.js", /\'UserChannel\'/
+  end
+
+  test "should generate channel with different resource name" do
+    run_generator ["my_name", "--broadcast-to", "--resource=under_scored_resource_name", "--no-stimulus"]
+
+    assert_file "app/channels/my_name_channel.rb" do |content|
+      assert_match /class\ MyNameChannel\ \</, content
+      assert_match /stream_for\ UnderScoredResourceName\.find\(params\[\:id\]\)/, content
+    end
+
+    assert_file "app/javascript/channels/my_name_channel.js", /"MyNameChannel"/
+    assert_no_file "app/javascript/controllers/my_name_controller.js"
+  end
+
+  test "should not generate stimulus controller if not requested" do
+    run_generator ["comment", "--broadcast-to", "--resource=comment", "--no-stimulus"]
+
+    assert_file "app/channels/comment_channel.rb"
+    assert_file "app/javascript/channels/comment_channel.js"
+    assert_no_file "app/javascript/controllers/comment_controller.js"
+  end
+
+  test "should run the generator when streaming from identifier" do
+    run_generator ["page", "--no-broadcast-to", "--identifier=page"]
+
+    assert_file "app/channels/page_channel.rb" do |content|
+      assert_match /PageChannel/, content
+      assert_match /stream_from\ \"page\"/, content
+    end
+
+    assert_file "app/javascript/channels/page_channel.js" do |content|
+      assert_match /"PageChannel"/, content
+      assert_match /import\ CableReady/, content
+      assert_match /if\ \(data\.cableReady\)\ CableReady\.perform\(data\.operations\)/, content
+    end
+  end
+
+  test "should run the generator when streaming without resource and different identifier" do
+    run_generator ["my_page", "--no-broadcast-to", "--identifier=ThisIsMyPage"]
+
+    assert_file "app/channels/my_page_channel.rb" do |content|
+      assert_match /MyPageChannel/, content
+      assert_match /stream_from\ \"this_is_my_page\"/, content
+    end
+
+    assert_file "app/javascript/channels/my_page_channel.js" do |content|
+      assert_match /"MyPageChannel"/, content
+      assert_match /import\ CableReady/, content
+      assert_match /if\ \(data\.cableReady\)\ CableReady\.perform\(data\.operations\)/, content
+    end
+  end
+
+  # some tests without generator options to simulate the inputs passed via cli
+
+  test "should generate channel with the same resource name and stimulus controller (without options)" do
+    CableReady::ChannelGenerator.any_instance.stubs(:using_broadcast_to?).returns(true)
```

Comment:
I'm good with the use of `mocha`, let's see what the others say 🙉

Reply:
Yep, just leard about it today. I'm happy to remove all mocha tests since they are basically a copy from the ones above to tests it without the options

Reply:
It’s a decent lib!